Add docs and sample code for sharing auth cookies between ASP.NET and ASP.NET Core (#28020)
dotnet/AspNetCore.Docs: aspnetcore/migration/inc/remote-authentication.md

Details on how to configure sharing auth cookies between ASP.NET and ASP.NET Core apps are available in [cookie sharing documentation](xref:security/cookie-sharing). The following samples in the [System.Web adapters](https://github.com/dotnet/systemweb-adapters) GitHub repo demonstrates remote app authentication with shared cookie configuration enabling both apps to sign users in and out :
  *([ASP.NET app](https://github.com/dotnet/systemweb-adapters/tree/main/samples/MvcApp)
  *  ASP.NET Core app](https://github.com/dotnet/systemweb-adapters/tree/main/samples/MvcCoreApp)

> Sharing authentication is a good option if the ASP.NET app is already using Microsoft.Owin cookie authentication and if it's possible to update the ASP.NET app and ASP.NET Core apps to use matching data protection settings (including having a shared location like a file path or Redis cache for storing data protection keys). For other scenarios, the remote authentication approach described previously in this doc is more flexible and will likely be a better fit.

Reviewer: ```suggestion
Sharing authentication is a good option if both the following are true:

* The ASP.NET app is already using `Microsoft.Owin` cookie authentication.
* It's possible to update the ASP.NET app and ASP.NET Core apps to use matching data protection settings. Matching shared data protection settings includes a shared file path, Redis cache, or Azure Blob Storage for storing data protection keys.

For other scenarios, the remote authentication approach described previously in this doc is more flexible and is probably a better fit.
```

1. Because Windows authentication depends on a handle to a Windows identity, Windows authentication is not supported by this feature. Future work is planned to explore how shared Windows authentication might work. See [dotnet/systemweb-adapters#246](https://github.com/dotnet/systemweb-adapters/issues/246) for more information.
1. This feature allows the ASP.NET Core app to make use of an identity authenticated by the ASP.NET app, but all actions related to users (logging on, logging off, etc.) still need to be routed through the ASP.NET app.

## Alternatives

> If authentication in the ASP.NET app is done using Katana Cookie Authentication Middleware, an alternative solution to sharing identity is to configure the ASP.NET and ASP.NET Core apps so that they are able to share an auth cookie. This allows both apps to determine user identity from the same cookie and signing in or out of one app will be reflected in the other. Note that because signing in typically depends on a specific database, shared cookie solutions will still typically have users always sign in through one of the apps (either ASP.NET or ASP.NET Core), but both apps will be able to see the users' identity and claims, and both apps will be able to log the user out.

Reviewer 2: > an alternative solution to sharing identity

If we're going to present alternative approaches, we probably need to provide guidance on when to use which one.

Author: Good point. I'll add something on that.

> If authentication in the ASP.NET app is done using Katana Cookie Authentication Middleware, an alternative solution to sharing identity is to configure the ASP.NET and ASP.NET Core apps so that they are able to share an auth cookie. This allows both apps to determine user identity from the same cookie and signing in or out of one app will be reflected in the other. Note that because signing in typically depends on a specific database, shared cookie solutions will still typically have users always sign in through one of the apps (either ASP.NET or ASP.NET Core), but both apps will be able to see the users' identity and claims, and both apps will be able to log the user out.

Reviewer: I put in tratchers change so you can apply his suggestion first, then mine, or just apply mine.

```suggestion
For authentication in the ASP.NET app using `Microsoft.Owin` Cookie Authentication Middleware, an alternative solution to sharing identity is configuring the ASP.NET and ASP.NET Core apps so that they are able to share an authentication cookie. Sharing an authentication cookie allows both apps to:

* Determine user identity from the same cookie.
* Sign a user in or out of one app and the sing in or out is reflected in the other app.

Note that because signing in typically depends on a specific database, shared cookie solutions:

* Typically have users always sign in through one of the apps, either ASP.NET or ASP.NET Core.
* Both apps are able to see the users' identity and claims.
* Both apps are able to sign the user out.
```

> Details on how to configure sharing auth cookies between ASP.NET and ASP.NET Core apps are available in [cookie sharing documentation](xref:security/cookie-sharing) and a sample is available in the System.Web adapters GitHub repo demonstrating remote app authentication (as described above) with shared cookie configuration so that both apps can sign users out ([sample ASP.NET app](https://github.com/dotnet/systemweb-adapters/tree/main/samples/MvcApp), [sample ASP.NET Core app](https://github.com/dotnet/systemweb-adapters/tree/main/samples/MvcCoreApp)).

Reviewer: ```suggestion
Details on how to configure sharing auth cookies between ASP.NET and ASP.NET Core apps are available in [cookie sharing documentation](xref:security/cookie-sharing). The following samples in the [System.Web adapters](https://github.com/dotnet/systemweb-adapters) GitHub repo demonstrates remote app authentication with shared cookie configuration enabling both apps to sign users in and out :
  *([ASP.NET app](https://github.com/dotnet/systemweb-adapters/tree/main/samples/MvcApp)
  *  ASP.NET Core app](https://github.com/dotnet/systemweb-adapters/tree/main/samples/MvcCoreApp)
```
